### DISCUSSION TITLE
HTML Report - Broken Links

### DISCUSSION THREAD

**Discussion by hpcnetworks** [sepinf-inc/IPED#2385] [Q&A]
Hi there!

I cannot access files attached to WhatsApp chats on HTML report. After clicking on images/videos/audio links, Chrome shows an error: _It may have been moved, edited, or deleted_.

Opening the source code on Chrome I can see the link: ../../B/B/BB3F77038FBE37158FBA162DBF0C63BBED49A3FC555C5231EE348A722D5FADC9.opus

The file, in fact is on filder B/B/, bit i think the problem is on ../../

Once the main page is on the report root folder, I think it should be _Exported/files/B/B/BB3F77038FBE37158FBA162DBF0C63BBED49A3FC555C5231EE348A722D5FADC9.opus_ , right?

Any idea?

Thanks in advance,

Bruno Costa

**Comment by wladimirleite** [comment]
When you open using IPEDSearchApp, does it work?

**Comment by lfcnassif** [comment]
If you are double clicking WhatsApp chats from the analysis UI to open them externally in your browser, those links are not expected to work. They should work internally in the application, and also when you generate an indexed report from the Options button on UI, last menu item. Those href links starting with ../../ are there so they will work when indexed reports are generated.

**Comment by hpcnetworks** [comment]
HI there!

I IPED SearchApp, when I click on audio files, it opens on Chrome, but when trying to open it on html page generated by IPED report tool, it says the link is broken. :(((

**Reply by hpcnetworks** [reply]
Sorry, not only audio files. Images as well...

**Comment by hpcnetworks** [comment]
Doing some research, I see the recursive directory is ok, the file is there, but the filename itself is a little bit different. I see 5 extra characters at the end of the filename. I think thats why the link is broken.

Please, check the attached images

Best Regards,

Bruno
![Audio 2](https://github.com/user-attachments/assets/7dc6ba65-7d14-4850-93b1-937096ae65d7)
![Audio 1](https://github.com/user-attachments/assets/b96d7507-97d6-4bc3-bba8-b15e11f9462f)

**Reply by lfcnassif** [reply]
Hum... Have you disabled md5 and sha1 hashes and left just sha256 enabled? That would explain it, it seems a bug when just sha256 is used. Iped has a file name clean up method that, beyond other things, truncates exported file names to 64 characters because of Joliet max file name size specification. Maybe we could increase that limit and ask users to burn optical medias with reports using UDF.

**Comment by hpcnetworks** [comment]
Yeah! Bingo! Only Sha256 is enabled.
I am reprocessing it right now. I will let you know when it finishes.

**Reply by lfcnassif** [reply]
Did it work?

**Reply by hpcnetworks** [reply]
Yes, It worked!!!

Thanks a Lot!!!

Unfortunetely, when I tried to generate the report with only thumbnails, when HTML links were generatedd without the extension .html. I Just generated It again with Full pica...

Thanks again.

**Reply by lfcnassif** [reply]
Please test the last 4.2.0 release, it includes a fix for #2203.
